nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/integration/provenance/ProvenanceEventsIT.java

```diff
@@ -420,4 +420,62 @@ public void testForkAutoGenerated() throws ExecutionException, InterruptedExcept
         final ProvenanceEventRecord thirdEvent = provRepo.getEvent(2L);
         assertEquals(ProvenanceEventType.DROP, thirdEvent.getEventType());
     }
+
+    @Test
+    public void testCloneOnMultipleConnectionsForRelationship() throws ExecutionException, InterruptedException, IOException {
+        final ProcessorNode generateProcessor = createGenerateProcessor(0);
+        final ProcessorNode passThroughProcessor = createProcessorNode((context, session) -> {
+            FlowFile original = session.get();
+            session.transfer(original, REL_SUCCESS);
+        }, REL_SUCCESS);
+
+        connect(generateProcessor, passThroughProcessor, REL_SUCCESS);
+        connect(passThroughProcessor, getTerminateProcessor(), REL_SUCCESS);
+        connect(passThroughProcessor, getTerminateAllProcessor(), REL_SUCCESS);
+
+        triggerOnce(generateProcessor);
+        triggerOnce(passThroughProcessor);
+
+        final ProvenanceEventRepository provRepo = getProvenanceRepository();
+        assertEquals(1L, provRepo.getMaxEventId().longValue());
+
+        final ProvenanceEventRecord firstEvent = provRepo.getEvent(0L);
+        assertEquals(ProvenanceEventType.CREATE, firstEvent.getEventType());
+
+        final ProvenanceEventRecord secondEvent = provRepo.getEvent(1L);
+        assertEquals(ProvenanceEventType.CLONE, secondEvent.getEventType());
+        assertEquals(1, secondEvent.getParentUuids().size());
+        assertEquals(1, secondEvent.getChildUuids().size());
+    }
+
+    @Test
+    public void testCloneOnMultipleConnectionsForRelationshipIncludesUpToDateAttributes() throws ExecutionException, InterruptedException, IOException {
+        final ProcessorNode generateProcessor = createGenerateProcessor(0);
+        final ProcessorNode passThroughProcessor = createProcessorNode((context, session) -> {
+            FlowFile original = session.get();
+            original = session.putAttribute(original, "test", "integration");
+
+            session.transfer(original, REL_SUCCESS);
+        }, REL_SUCCESS);
+
+        connect(generateProcessor, passThroughProcessor, REL_SUCCESS);
+        connect(passThroughProcessor, getTerminateProcessor(), REL_SUCCESS);
+        connect(passThroughProcessor, getTerminateAllProcessor(), REL_SUCCESS);
+
+        triggerOnce(generateProcessor);
+        triggerOnce(passThroughProcessor);
+
+        final ProvenanceEventRepository provRepo = getProvenanceRepository();
+        assertEquals(1L, provRepo.getMaxEventId().longValue());
```

Comment:
After applying this PR, I run the similar flow, GenerateFlowFile -> UpdateAttribute -> LogAttribute:
![image](https://user-images.githubusercontent.com/1107620/56410937-f1665c80-62b9-11e9-9eb0-32611a05f545.png)

And 3 provenance events were created:
![image](https://user-images.githubusercontent.com/1107620/56410960-06db8680-62ba-11e9-91c6-f223d329bc08.png)

However, this test case expects only two events are created. This looks different from what actually happens with a live NiFi flow. Also, you used `thirdEvent` below, I wonder if you were expecting 3 events with this?

Reply:
@ijokarumawak in the example that you have above, you are getting the ATTRIBUTES MODIFIED event because UpdateAttribute explicitly creates that event - not the framework. So if you used, for example, LogAttribute instead of UpdateAttribute, then you should not see that behavior.
As for the `thirdEvent` that was just a mistake because I copied & pasted the assertions of the test above it. Will fix that.